### PR TITLE
fix: Hide poap block if there is no POAP is setup

### PR DESCRIPTION
**Pull request by ChaituVR**
### Issues

Fixes https://github.com/snapshot-labs/pitches/issues/14

### Changes 
*_(Briefly describe the changes made in this PR)_*
1. Hides the poap block if Poap is not setup for the proposal
2. There is no way to hide the 404 error message, so i am gonna ignore it for now

### How to test:
- POAP block should not show up here https://snapshot-git-hide-poap-if-no-poap-snapshot.vercel.app/#/stgdao.eth/proposal/0x108a9e597560c4f249cd8be23acd409059fcd17bb2290d69a550ac2232676e7d
- POAP block should show up here 
https://snapshot-git-hide-poap-if-no-poap-snapshot.vercel.app/#/fractiontoken.eth/proposal/0xeaaf4b685084762f952f587fc8aa0725b40bc4c2d77e4e976a4125d3621cf7c3

### Self-review checklist
- [x] I have performed a full self-review of my changes
- [x] I have tested my changes on a preview deployment
- [ ] I have tested my changes on different screen sizes (sm, md)
- [ ] I have tested my changes on a custom domain

### Additional notes or considerations
*_(Include any other relevant information or context that may be helpful for the reviewer)_*

### CODE REVIEW

File: src/plugins/poap/components/CustomBlock.vue [hunk 1]
```diff
@@ -147,7 +147,11 @@ export default {
 </script>
 
 <template>
-  <BaseBlock title="I voted POAP" :loading="loading">
+  <BaseBlock
+    title="I voted POAP"
+    :loading="loading"
+    v-if="currentState !== 'NO_POAP'"
```
Comment (samuveth) [inline src/plugins/poap/components/CustomBlock.vue:153]:
```suggestion
    v-if="currentState !== 'NO_POAP'"
    title="I voted POAP"
    :loading="loading"
```

`v-if` should come first

Comment (samuveth) [reply src/plugins/poap/components/CustomBlock.vue:153]: Also with this change the loading is redundant, as the block will not be visible until loading is complete. I'm not sure this is the best approach

**Comment by wa0x6e** [comment]
Should we also display another POAP block only for space controller, to invite them to setup the POAP plugin ? (Similar to SnapIt!)

File: src/plugins/poap/components/CustomBlock.vue (hunk 1 above)
Comment (ChaituVR) [reply src/plugins/poap/components/CustomBlock.vue:153]: What do you think is the best approach then?

We should show loading and then hide?
Or we remove loading completely?

**Comment by ChaituVR** [comment]
> Should we also display another POAP block only for space controller, to invite them to setup the POAP plugin ? (Similar to SnapIt!)

Nice idea but better to ask Fab or Guillem  on  pitch document or discord

File: src/plugins/poap/components/CustomBlock.vue (hunk 1 above)
Comment (ChaituVR) [reply src/plugins/poap/components/CustomBlock.vue]: Now it displays loading, if POAP doesn't exist we hide the block, else keep it

Comment (samuveth) [reply src/plugins/poap/components/CustomBlock.vue]: If the Poap plugin is setup for the space we should always show the POAP block and loading on the proposal. And if there is no POAP for the specific proposal then it should show a message inside the block. @ChaituVR

Comment (ChaituVR) [reply src/plugins/poap/components/CustomBlock.vue]: But this is the already current functionality 😄 It is confusing that we show a message on proposals that don't have POAP. so we wanted to hide no? 🤔

Comment (ChaituVR) [reply src/plugins/poap/components/CustomBlock.vue]: Could you explain your concerns with pitch owners on Discord plz? so they can take a decision

Comment (minimayki6232) [reply src/plugins/poap/components/CustomBlock.vue]: > ```suggestion
>     v-if="currentState !== 'NO_POAP'"
>     title="I voted POAP"
>     :loading="loading"
> ```
> 
> `v-if` should come first

**Pull request merged**